conjure-go-client/httpclient/internal/request_retrier.go

```diff
@@ -79,44 +66,45 @@ func (r *RequestRetrier) attemptsRemaining() bool {
 	return r.attemptCount < r.maxAttempts
 }
 
-// GetNextURI returns the next URI a client should use, or an error if there's no suitable URI.
-// This should only be called after validating that there's a suitable URI to use via ShouldGetNextURI, in which case
-// an error will never be returned.
-func (r *RequestRetrier) GetNextURI(ctx context.Context, resp *http.Response, respErr error) (string, error) {
+// GetNextURI returns the next URI a client should use, or empty string if no suitable URI remaining to retry.
+// isRelocated is true when the URI comes from a redirect's Location header. In this case, it already includes the request path.
+func (r *RequestRetrier) GetNextURI(resp *http.Response, respErr error) (uri string, isRelocated bool) {
 	defer func() {
 		r.attemptCount++
 	}()
 	if r.attemptCount == 0 {
-		return r.removeMeshSchemeIfPresent(r.currentURI), nil
-	} else if !r.ShouldGetNextURI(resp, respErr) {
-		return "", r.getErrorForUnretriableResponse(ctx, resp, respErr)
+		// First attempt is always successful
+		r.retrier.Next() // trigger first retry so later calls have backoff
```

Comment:
I added this so the second request (first retry) gets delayed

Reply:
So basically this will not block on the first request since the `isReset` will be true and then subsequent retry requests will then block for the appropriate time